app.py

```diff
@@ -1,14 +1,92 @@
 from flask import Flask, request, render_template
 import redis
+import psycopg2
+from psycopg2.extensions import AsIs
 import json
 import logging
 
 logger = logging.getLogger(__name__)
 
-r = redis.Redis(host='172.17.0.1', port=6379)
-
 app = Flask(__name__)
 
+def initialize():
+    global p
+    global r
+    try:
+        p = psycopg2.connect(user="postgres",
+                            password="postgres",
+                            host="172.17.0.1",
+                            port="5432",
+                            database="hello")
+
+        r = redis.Redis(host='172.17.0.1', port=6379)
+    except (Exception) as e:
+        return False
+    return True
+
+@app.route("/users", methods = ['POST', 'GET'])
+def users():
+
+    # If both are specified, id takes precedence
+    if request.method == 'GET':
+
+        c = p.cursor()
+
+        id   = request.args.get('id')
+        name = request.args.get('name')
+
+        try:
+            if id:
+                c.execute("SELECT * FROM users WHERE id = %s;", (id,))
+                record = c.fetchmany()
+            elif name and not id:

```

Comment:
you don't even need the `and not id` here as its covered by the previous `if`, and this statement is in an `elif`